sentry/src/routes/channel.rs

```diff
@@ -43,22 +45,32 @@ pub async fn create_channel<A: Adapter>(
 ) -> Result<Response<Body>, ResponseError> {
     let body = hyper::body::to_bytes(req.into_body()).await?;
 
-    let channel = serde_json::from_slice::<Channel>(&body)?;
+    let channel = serde_json::from_slice::<Channel>(&body)
+        .map_err(|e| ResponseError::FailedValidation(e.to_string()))?;
```

Comment:
Wait what?! Why is deserialization a FailedValidation, but `validate_channel` result is only BadRequest?

Reply:
yeah, the difference is in the responses sent to the request client

Reply:
It's a bit confusing reading the code like this, would be good to change it to remove confusion.

Reply:
No its not, `validate_channel` is not `FailedValidation` because its not checking the request body. `FailedValidation` is the error response  for checking  the request body

Reply:
BadRequest is for when you pass in valid request body but during processing the request something is wrong